captioning/utils/image_preproc.py

```diff
@@ -105,29 +109,24 @@ def process_feature_extraction(output,
         boxes_list.append(max_box[keep_boxes])
         classes_list.append(max_cls[keep_boxes])
 
-    return boxes_list, feat_list, classes_list, conf_list
+    return [boxes_list, feat_list, classes_list, conf_list]
 
 
-# Given a single/list of image(s) returns features, bboxes, scores and classes
-def get_detectron_features(image_path,
+# Given a list of image(s) returns features, bboxes, scores and classes
+def get_detectron_features(image_paths,
                            detection_model,
                            get_boxes,
                            feat_name,
-                           batch_mode=False):
+                           device):
     img_tensor, im_scales = [], []
 
-    if batch_mode:
-        # TODO: Use better arg-name and loop vars
-        for img_path in image_path:
-            im, im_scale = image_transform(img_path)
-            img_tensor.append(im)
-            im_scales.append(im_scale)
-    else:
-        im, im_scale = image_transform(image_path)
-        img_tensor, im_scales = [im], [im_scale]
+    for img_path in image_paths:
+        im, im_scale = image_transform(img_path)
```

Comment:
Can we change this to
```
im = read_image(img_path)
im, im_scale = transform_image(im)
```
Feels odd for `transform` to also be reading the image inside it.

Reply:
Done!